File: test_2.py
def func_2(array):

  min = array[0]
  for el in array: #поиск минимального числа в массиве
    if el < min:
      min = el

  common = [] #список общих делителей
  for i in range(2, min+1): #числа от 2х до минимального числа
    if all(el % i == 0 for el in array): #проверка на делимость каждого элемента массива на число
      common.append(i) #добавление в список

  return common

print(func_2(array = list(map(int, input().split()))))

File: test_2.js
const readline = require('readline').createInterface( {
    input: process.stdin,
    output: process.stdout,
  });

function func_2(array) {

  let min = array[0];
  for (let i=0; i< array.length; i++) {
    if (array[i] < min) {
      min = array[i];
    }
}

  let common = [];
  for (let i = 2; i<= min; i++) { 
    let divisibility = true;
    for (let j=0; j< array.length; j++) {
        if (array[j] % i !== 0) {
            divisibility = false;
            break;
        }
    }
    if (divisibility) {
        common.push(i);
    }
}
  return common;
}

readline.question("Введите числа через пробел: ", function(input) {
    const array = input.split(" ").map(function(item) {
        return parseInt(item);
    });
    console.log(func_2(array));
    readline.close();
})